tests/test_async_s3_bucket.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of ts_salobj.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import io
+import os
+
+import asynctest
+
+import boto3
+import moto
+
+from lsst.ts import salobj
+
+
+class AsyncS3BucketTest(asynctest.TestCase):
+    def setUp(self):
+        # Use moto.mock_s3 in "raw mode" because:
+        # * @moto.mock_s3() does not work on async test methods.
+        # * Even if it did, or if I used moto.mock_s3() as a context manager,
+        #   I'd have to make the s3 bucket inside the test code.
+        self.mock = moto.mock_s3()
+        self.mock.start()
```

Comment:
You have the same `file_data`, `key`, and `bucket` in each test: is it worth defining those in `setUp` and using the `self` versions in the tests?